hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceTest.java

```diff
@@ -0,0 +1,618 @@
+        IScheduledFuture<Double> first = executorService.schedule(
+                named(taskName, new PlainCallableTask()), delay, TimeUnit.SECONDS);
+
+        ScheduledTaskHandler handler = first.getHandler();
+        IScheduledFuture<Double> copy = executorService.getScheduledFuture(handler);
+
+        assertEquals(first, copy);
+    }
+
+    @Test
+    public void getAllScheduled()
+            throws ExecutionException, InterruptedException {
+
+        HazelcastInstance[] instances = createClusterWithCount(3);
+        IScheduledExecutorService s = instances[0].getScheduledExecutorService("s");
+        s.scheduleOnAllMembers(new PlainCallableTask(), 0, TimeUnit.SECONDS);
+
+        Set<Member> members = instances[0].getCluster().getMembers();
+        Map<Member, List<IScheduledFuture<Double>>> allScheduled = s.getAllScheduledFutures();
+
+        assertEquals(members.size(), allScheduled.size());
+
+        for (Member member : members) {
+            assertEquals(1, allScheduled.get(member).size());
+            assertEquals(25.0, allScheduled.get(member).get(0).get(), 0);
+        }
+    }
+
+    public HazelcastInstance[] createClusterWithCount(int count) {
+        HazelcastInstance[] instances = new HazelcastInstance[count];
+
+        factory = createHazelcastInstanceFactory(count);
+        for (int i = 0; i < count; i++) {
+            instances[i] = factory.newHazelcastInstance();
+        }
+
+        return instances;
+    }
+
+    static class StatefullRunnableTask
+            implements Runnable, Serializable,
+                       HazelcastInstanceAware, StatefulTask<String, Integer> {
+
+        final String latchName;
+
+        int status;
+
+        transient HazelcastInstance instance;
+
+        StatefullRunnableTask(String latchName, HazelcastInstance instance) {
+            this.latchName = latchName;
+            this.instance = instance;
+        }
+
+        @Override
+        public void run() {
+            status = 66 * 77;
+        }
+
+        @Override
+        public void load(Map<String, Integer> snapshot) {
+            if (!snapshot.isEmpty()) {
+                assertEquals(66 * 77, (int) snapshot.get("status"));
+                instance.getCountDownLatch(latchName).countDown();
+            }
+        }
+
+        @Override
+        public void save(Map<String, Integer> snapshot) {
+            snapshot.put("status", status);
+        }
+
+        @Override
+        public void setHazelcastInstance(HazelcastInstance hazelcastInstance) {
+            this.instance = hazelcastInstance;
+        }
+    }
+
+    static class ICountdownLatchRunnableTask implements Runnable, Serializable, HazelcastInstanceAware {
+
+        final String latchName;
+
+        transient HazelcastInstance instance;
+
+        ICountdownLatchRunnableTask(String latchName, HazelcastInstance instance) {
+            this.latchName = latchName;
+            this.instance = instance;
```

Comment:
This is not the case for local-member invocation, it is when its a remote call, and the OP has to get deserialized, but not when local